File: app/models/coach.rb
class Coach < ActiveRecord::Base
  attr_accessible :name, :about, :races, :servers, :languages
  validates_presence_of :name, :about, :races, :servers, :languages

  def main_server_icon
    "coaches_#{servers.first}.png"
  end

  def race_icons
    races.map { |race| "race_#{race}_icon.png" }
  end

end

File: app/assets/javascripts/models/coach.js
Scvrush.Coach = DS.Model.extend({
  name: DS.attr("string"),
  about: DS.attr("string"),

  races: DS.attr("object"),
  servers: DS.attr("object"),
  languages: DS.attr("object"),

  primaryServer: function() {
    return this.get("servers.firstObject");
  }.property("servers.[]"),

  primaryServerIcon: function() {
    return "/assets/coaches_" + this.get("primaryServer") + ".png";
  }.property("primaryServer"),

  raceIcon: function(race) {
    if (race) {
      return "/assets/race_" + race.toLowerCase() + "_icon.png";
    } else {
      return "/assets/race_random_icon.png";
    }
  },

  raceIcons: function() {
    var self = this;

    return this.get("races").map(function(race) {
      return self.raceIcon(race);
    });
  }.property("races.@each"),

  languageFlags: function() {
    return this.get("languages").map(function(language) {
      return "/assets/flags/" + language + ".png";
    });
  }.property("languages.@each"),

})
